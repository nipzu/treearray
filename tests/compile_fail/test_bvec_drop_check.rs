fn main() {
    let mut x = btreevec::BTreeVec::<&str>::new();
    {
        let s = String::from("Hello!");
        x.push_back(s.as_str());
    }
}