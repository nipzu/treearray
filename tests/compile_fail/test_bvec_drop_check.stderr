error[E0597]: `s` does not live long enough
 --> tests/compile_fail/test_bvec_drop_check.rs:5:21
  |
5 |         x.push_back(s.as_str());
  |                     ^^^^^^^^^^ borrowed value does not live long enough
6 |     }
  |     - `s` dropped here while still borrowed
7 | }
  | - borrow might be used here, when `x` is dropped and runs the `Drop` code for type `BTreeVec`
  |
  = note: values in a scope are dropped in the opposite order they are defined
